compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/hoist_for_affine_apply.mlir

```diff
@@ -0,0 +1,43 @@
+// RUN: iree-opt --split-input-file --pass-pipeline="builtin.module(func.func(iree-amdaie-hoist-for-affine-apply))" %s | FileCheck %s
+
+// CHECK-LABEL: @hoist_affine_map
+// CHECK-SAME:  %[[LB1:.+]]: index, %[[UB1:.+]]: index, %[[STEP1:.+]]: index, %[[LB2:.+]]: index, %[[UB2:.+]]: index, %[[STEP2:.+]]: index
+#map = affine_map<(d0) -> (d0 * 32)>
+#map1 = affine_map<(d0) -> (d0 * 64)>
+func.func @hoist_affine_map(%lb1: index, %ub1: index, %step1: index, %lb2: index, %ub2: index, %step2: index) {
+  scf.for %i = %lb1 to %ub1 step %step1 {
+    scf.for %j = %lb2 to %ub2 step %step2 {
+      %0 = affine.apply #map(%i)
+      %1 = affine.apply #map1(%i)
+    }
+  }
+  // CHECK:     scf.for %[[IV1:.+]] = %[[LB1]] to %[[UB1]] step %[[STEP1]]
+  // CHECK:       %[[MAP0:.+]] = affine.apply #map(%[[IV1]])
+  // CHECK:       %[[MAP1:.+]] = affine.apply #map1(%[[IV1]])
+  // CHECK:       scf.for %[[IV2:.+]] = %[[LB2]] to %[[UB2]] step %[[STEP2]]
+  // CHECK-NOT:     %[[MAP2:.+]] = affine.apply #map(%[[IV1]])
+  // CHECK-NOT:     %[[MAP3:.+]] = affine.apply #map1(%[[IV1]])
```

Comment:
I guess I didn't explain properly - the above snippet I suggested was for the part where the `affine.apply` is hoisted (i.e. between the loops).

For sake of completeness, this is what you should be doing instead :-
```
  //      CHECK:     scf.for %[[IV1:.+]] = 
  // CHECK-NEXT:       affine.apply #map(%[[IV1]])
  // CHECK-NEXT:       affine.apply #map1(%[[IV1]])
  // CHECK:          scf.for
  // CHECK-NOT:        affine.apply #map(%[[IV1]])
  // CHECK-NOT:        affine.apply #map(%[[IV1]])
```
(The above assumes that NO other op is being created by the pass between the two loops, hence `CHECK-NEXT`)

To reiterate what I was trying to suggest here is - always check the following while creating lit test to make it as concise as possible :- 
1. Are the SSA values being matched used anywhere? If not, we might as well move on to the next important/relevant match.
2. What other extra matching can I let go of which isn't communicating the main intention of the pass?

Let me know if that makes sense.

Reply:
I think CHECK-NEXT would be wrong here because it implies that the particular check is on the next immediate line, however the pass does not guarantee that and it doesn't need to, that is either order is correct sc.for -> map -> map1  or sc.for-> map1 -> map, so just using CHECK is correct here as we dont want the test to fail in case that order changes in the future.

Reply:
Ya i think its fine this way too.

Reply:
@nirvedhmeshram  I wrote my answer before I saw yours as I didn't refresh the page. I think you make a good point on `CHECK-NEXT` 

> that is either order is correct sc.for -> map -> map1 or sc.for-> map1 -> map

I think `CHECK` will also check for the order, so `scf.for-> map1 -> map` will fail.

Based on the discussion, I updated to `CHECK-DAG` to allow for a different order as well.

Reply:
oh you are right, CHECK-DAG is most suitable here.